projects/js-packages/publicize-components/src/components/unified-modal/social-post-preview/connection-panels.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+import { useAnalytics } from '@automattic/jetpack-shared-extension-utils';
+import {
+	Flex,
+	FormToggle,
+	__experimentalGrid as Grid,
+	Panel,
+	PanelBody,
+	PanelRow,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import useSocialMediaConnections from '../../../hooks/use-social-media-connections';
+import { Connection } from '../../../social-store/types';
+import {
+	getA11yLabelForConnectionPreview,
+	getA11yLabelForConnectionToggle,
+} from '../../../utils/misc';
+import ConnectionIcon from '../../connection-icon';
+import { useConnectionState } from '../../form/use-connection-state';
+import { PostPreview } from '../../social-post-modal/post-preview';
+import styles from './styles.module.scss';
+
+/**
+ * Connection Panels component for the social preview modal.
+ *
+ * @return - Connection Panels component.
+ */
+export function ConnectionPanels() {
+	const { recordEvent } = useAnalytics();
+	const { toggleById } = useSocialMediaConnections();
+	const { connections } = useSocialMediaConnections();
```

Comment:
We could merge these 
```suggestion
    const { toggleById, connections } = useSocialMediaConnections();
```